lib/ecmascript.ts

```diff
@@ -1307,12 +1336,13 @@ export function PrepareTemporalFields<
   RequiredFields extends ReadonlyArray<FieldKeys> | FieldCompleteness
 >(
   bag: Partial<Record<FieldKeys, unknown>>,
-  fields: ReadonlyArray<FieldKeys>,
+  fields: Array<FieldKeys>,
```

Comment:
Does this break type inference for the return type?

If it does, then I'd be tempted to lie about the type inside this function and pretend it is still readonly outside.

For the most part the field ordering probably doesn't matter (in the type-system) in every other call, and it is unfortunate there isn't a good way to model that in the typesystem.

Reply:
The return type is still inferred. What it \*does\* do is allow callers to call it with a non-readonly array, which might make them forget to put `as const` on the array input. But if the caller types the input properly then the output will get the correct inferred type. I admit I am a little nervous about doing mutation on an input declared to be readonly, but also fine to do the lying thing too. 

How do you want to proceed?

Reply:
My recollection from working on the PrepareTemporalFields typing is that if the inference degrades a bunch of stuff will break loudly, so it shouldn't be a problem in the future so long as it still happens most of the time now. Let's leave this as-is and we can always fix it up later if we need to by either breaking the readonly contract here or possibly by using a sorted copy here somehow.